polaris-core/src/main/java/org/apache/polaris/core/persistence/bootstrap/RootCredentialsSet.java

```diff
@@ -130,18 +137,29 @@ static RootCredentialsSet fromList(List<String> credentialsList) {
    * }
    * </pre>
    */
-  static RootCredentialsSet fromUrl(URL url) {
+  static RootCredentialsSet fromUri(URI uri) {
+    Preconditions.checkNotNull(uri);
+    Preconditions.checkArgument(
+        Strings.isNullOrEmpty(uri.getHost()),
+        "Remote URIs are not allowed for RootCredentialsSet: %s",
+        uri);
```

Comment:
Actually thinking a bit more about this, you can still get a "host" via a `jar` URI like `jar:http://foo.bar/...` or all the `file` scheme "specialties". But that's rather a question what source locations we want to allow.

For the scope of this PR, the change as it is looks fine.